API/Controllers/CountryController.cs

```diff
@@ -29,6 +29,51 @@ public IEnumerable<CountryContract> GetAllCountries()
             return countryContracts;
         }
 
+        [HttpGet()]
+        public IEnumerable<string> GetAllCountryNames()
+        {
+            var countries = _countryRepository.GetAllCountries();
+
+            var countryNames = new List<string>();
+
+            foreach(var country in countries)
+            {
+                countryNames.Add(country.Name);
+            }
+
+            return countryNames;
+        }
+
+        [HttpGet()]
+        public IEnumerable<string> GetAllCountryCodes()
+        {
+            var countries = _countryRepository.GetAllCountries();
+
+            var countryCodes = new List<string>();
+
+            foreach (var country in countries)
+            {
+                countryCodes.Add(country.Code);
+            }
+
+            return countryCodes;
+        }
+
+        [HttpGet()]
+        public IDictionary<string,string> GetAllCountryNamesDict()
```

Comment:
Is this dictionary needed or could we just return a list of countries without data?

Reply:
Fixed